packages/remote-server/src/app.ts

```diff
@@ -11,23 +11,30 @@ config();
 
 class App {
 	public express: express.Application;
-	private server: AgentMcpServer;
+	private static serverInstance: AgentMcpServer | null = null;
 
 	constructor() {
 		this.express = express();
 		this.middleware();
 		this.routes();
-		const tools = process.env.TOOLS ? normalizeEnvVar(process.env.TOOLS).split(',') : [];
-		this.server = new AgentMcpServer({
-			tools,
-			version: '1.0.0',
-			credo: {
-				port: parseInt(process.env.CREDO_PORT || '3000', 10),
-				domain: process.env.CREDO_ENDPOINT,
-				name: process.env.CREDO_NAME,
-				cosmosPayerSeed: process.env.CREDO_CHEQD_TESTNET_MNEMONIC,
-			},
-		});
+		// Instantiate singleton AgentMcpServer instance
```

Comment:
@sownak I added a simple fix here https://github.com/cheqd/mcp-toolkit/pull/25. Can you take a look